src/pages/RFOX/components/Stake/StakeInput.tsx

```diff
@@ -25,27 +42,187 @@ const formControlProps = {
+    [cryptoAmount, fiatAmount],
+  )
+
+  const { data: cooldownPeriod, isLoading: isCooldownPeriodLoading } = useContractRead({
+    abi: foxStakingV1Abi,
+    // TODO(gomes): const somewhere
+    address: '0x0c66f315542fdec1d312c415b14eef614b0910ef',
+    functionName: 'cooldownPeriod',
+    chainId: arbitrum.id,
+    staleTime: Infinity,
+    select: data => formatDuration(Number(data as BigInt)),
+  })
+
+  const callData = useMemo(() => {
+    if (!(isValidStakingAmount && runeAddress)) return
+
+    return encodeFunctionData({
+      abi: foxStakingV1Abi,
+      functionName: 'stake',
+      args: [BigInt(cryptoAmount), runeAddress],
+    })
+  }, [cryptoAmount, isValidStakingAmount, runeAddress])
+
+  const estimateFeesInput = useMemo(
+    () => ({
+      // This is a contract call i.e 0 value
+      amountCryptoPrecision: '0',
+      assetId: asset?.assetId ?? '',
+      feeAssetId: feeAsset?.assetId ?? '',
+      // TODO(gomes): const somewhere
+      to: '0x0c66f315542fdec1d312c415b14eef614b0910ef',
+      sendMax: false,
+      memo: callData,
+      accountId: stakingAssetAccountId ?? '',
+      contractAddress: undefined,
+    }),
+    [asset?.assetId, callData, feeAsset?.assetId, stakingAssetAccountId],
+  )
+
+  const { data: allowanceDataCryptoBaseUnit, isSuccess: isAllowanceDataSuccess } = useAllowance({
+    assetId: asset?.assetId,
+    // TODO(gomes): const somewhere
+    spender: '0x0c66f315542fdec1d312c415b14eef614b0910ef',
+    from: stakingAssetAccountId ? fromAccountId(stakingAssetAccountId).account : undefined,
+  })
+
+  const allowanceCryptoPrecision = useMemo(() => {
+    if (!allowanceDataCryptoBaseUnit) return
+    if (!asset) return
+
+    return fromBaseUnit(allowanceDataCryptoBaseUnit, asset?.precision)
+  }, [allowanceDataCryptoBaseUnit, asset])
+
+  // TODO(gomes): balance checks too
+  const isApprovalRequired = useMemo(
+    () => bnOrZero(allowanceCryptoPrecision).lt(cryptoAmount),
+    [allowanceCryptoPrecision, cryptoAmount],
+  )
+
+  const isEstimatedFeesEnabled = useMemo(
+    () =>
+      Boolean(
+        isValidStakingAmount &&
+          asset &&
+          runeAddress &&
+          isAllowanceDataSuccess &&
+          !isApprovalRequired,
+      ),
+    [asset, isAllowanceDataSuccess, isApprovalRequired, isValidStakingAmount, runeAddress],
+  )
+
+  // TODO(gomes): move this queryFn out of lending
+  const estimatedFeesQueryKey: EstimatedFeesQueryKey = useMemo(
+    () => [
+      'estimateFees',
+      {
+        enabled: isEstimatedFeesEnabled,
+        asset,
+        assetMarketData: assetMarketDataUserCurrency,
+        feeAsset,
+        feeAssetMarketData,
+        estimateFeesInput,
+      },
+    ],
+    [
+      asset,
+      assetMarketDataUserCurrency,
+      estimateFeesInput,
+      feeAsset,
+      feeAssetMarketData,
+      isEstimatedFeesEnabled,
+    ],
+  )
+
+  const { data: estimatedFees, isLoading: isEstimatedFeesLoading } = useQuery({
```

Comment:
And we'll want to move this query to react-queries, this is a super common use-case that we keep on duplicating with very small logic changes per domain with wrappers instead of having a base reusable query

Reply:
Done with a new `getEvmFee` query